### DISCUSSION TITLE
Fails to generate STOCKINFO

### DISCUSSION THREAD

**Discussion by andrebeliveau** [csingley/ofxtools#146] [Q&A] [answered]
I am trying to convert quicken investments statements to OFX.  I tried to generate the STOCKINFO section of a response and I keep getting the error: TypeError: SECLIST can't contain list as list item

I am able to generate the investment and bank transactions, but I have a problem to correctly program the STOCKINFO part.
Here is a code example that generates the error.

Thanks for your help.
---------
import ofxtools.models as m

def stockinfoTest():

    stockinfo = []
    status = m.STATUS(code=0, severity='INFO')
    ofxstring=m.STOCKINFO(
        secinfo=m.SECINFO(
            secid=m.SECID(uniqueid='FB',uniqueidtype='TICKER'), 
            secname='Facebook'
        )
    )
    stockinfo.append(ofxstring)

    ofxstring=m.STOCKINFO(
        secinfo=m.SECINFO(
            secid=m.SECID(uniqueid='AMEX',uniqueidtype='TICKER'), 
            secname='American Express'
        )
    )
    stockinfo.append(ofxstring)

    seclistmsgsrsv1 = m.SECLISTMSGSRSV1(
            seclist=m.SECLIST(stockinfo)
    )

    return seclistmsgsrsv1

stockinfoTest()

**Comment by csingley** [comment]
The syntax for creating an OFX list-type aggregate in `ofxtools` doesn't look like that, sorry.  Think flat, not nested.

When instantiatiating a `SECLIST`, you can't feed it a list (i.e. your `stockinfo`).  Instead you need to just feed it all the contained `STOCKINFO` subaggregates as positional `*arg`s of the `__init__()` function.

So first create `stockinfo1` and `stockinfo2`, then say

```
seclistmsgsrsv1 = m.SECLISTMSGSRSV1(
        seclist=m.SECLIST(stockinfo1, stockinfo2)
)
```

**Reply by andrebeliveau** [reply]
Thank you for your prompt reply.

**Reply by andrebeliveau** [reply]
I still get the same error with the proposed code:
import ofxtools.models as m

def stockinfoTest():

    stockinfo1=m.STOCKINFO(
        secinfo=m.SECINFO(
            secid=m.SECID(uniqueid='FB',uniqueidtype='TICKER'), 
            secname='Facebook'
        )
    )

    stockinfo2=m.STOCKINFO(
        secinfo=m.SECINFO(
            secid=m.SECID(uniqueid='AMEX',uniqueidtype='TICKER'), 
            secname='American Express'
        )
    )

    seclistmsgsrsv1 = m.SECLISTMSGSRSV1(
        seclist=m.SECLIST(stockinfo1, stockinfo2)
    )
    return seclistmsgsrsv1

stockinfoTest()

**Comment by andrebeliveau** [comment, marked as answer]
It worked once I used the code below.  It allows me to keep a single list of all stocks.

def stockinfoTest():
    stockinfo = []

    ofxstring=m.STOCKINFO(
        secinfo=m.SECINFO(
            secid=m.SECID(uniqueid='FB',uniqueidtype='TICKER'), 
            secname='Facebook'
        )
    )
    stockinfo.append(ofxstring)

    ofxstring=m.STOCKINFO(
        secinfo=m.SECINFO(
            secid=m.SECID(uniqueid='AMEX',uniqueidtype='TICKER'), 
            secname='American Express'
        )
    )
    stockinfo.append(ofxstring)

    seclistmsgsrsv1 = m.SECLISTMSGSRSV1(
        m.SECLIST(*stockinfo)
    )

    return seclistmsgsrsv1

**Comment by csingley** [comment]
> I still get the same error with the proposed code

You probably get a different error asking you to pass `SECLIST` as a positional argument, not a keyword argument.  See the class definition in `ofxtools.models.invest.msgsets`:

```
class SECLISTMSGSRSV1(Aggregate):
    <snip />
    seclisttrnrs = ListAggregate(SECLISTTRNRS)
    seclist = ListAggregate(SECLIST)
```

**Reply by csingley** [reply]
The general syntax is to pass list elements as positional arguments, and pass unique subaggregates as keyword arguments.
